packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx

```diff
@@ -0,0 +1,202 @@
+import { formatTimestamp } from '@lightdash/common';
+import {
+    ActionIcon,
+    Alert,
+    Button,
+    CopyButton,
+    Modal,
+    Select,
+    Stack,
+    TextInput,
+    Title,
+    Tooltip,
+} from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { IconAlertCircle, IconCheck, IconCopy } from '@tabler/icons-react';
+import { FC, useEffect, useState } from 'react';
+import useHealth from '../../../hooks/health/useHealth';
+import { useCreateAccessToken } from '../../../hooks/useAccessToken';
+import MantineIcon from '../../common/MantineIcon';
+
+export const CreateTokenModal: FC<{
+    onBackClick: () => void;
+}> = ({ onBackClick }) => {
+    const health = useHealth();
+
+    const {
+        data,
+        mutate: createAccessToken,
+        isLoading,
+        isSuccess,
+    } = useCreateAccessToken();
+
+    const [expireDate, setExpireDate] = useState<Date | undefined>();
+
+    const form = useForm({
+        initialValues: {
+            description: '',
+            expiresAt: '',
+        },
+    });
+
+    const expireOptions = [
+        {
+            label: 'No expiration',
+            value: '',
+        },
+        {
+            label: '7 days',
+            value: '7',
+        },
+        {
+            label: '30 days',
+            value: '30',
+        },
+        {
+            label: '60 days',
+            value: '60',
+        },
+        {
+            label: '90 days',
+            value: '90',
+        },
+    ];
+
+    useEffect(() => {
+        if (isSuccess) {
+            setExpireDate(data.expiresAt);
+        }
+    }, [data?.expiresAt, isSuccess]);
+
+    const handleOnSubmit = form.onSubmit(({ description, expiresAt }) => {
+        const currentDate = new Date();
+        const dateWhenExpires = !!Number(expiresAt)
+            ? new Date(
+                  currentDate.setDate(
+                      currentDate.getDate() + Number(expiresAt),
+                  ),
+              )
+            : undefined;
+        setExpireDate(dateWhenExpires);
```

Comment:
do we need this state? especially on submit and I see it clears up after successful query

Reply:
🤦 we do not!